### ISSUE TITLE
The Installer does not install all neccesary files in order for the Dial App to function correctly.

### ISSUE THREAD

**Issue by gantonjo-tnm** [ahmsville/Ahmsville-Dial-App#1]
![image](https://user-images.githubusercontent.com/42378451/134235250-38e74558-8ecc-4643-b694-ca755c7d39a4.png)
As can be seen in the picture, the folders related to the Arduino project files are not created by the Dial App, hence the Dial App crashes. Tried to install version V1 and the latest version, but both versions behaves equal.

**Comment by ahmsville** [comment]
Hi @gantonjo-tnm , have you installed the Ardunio ide?, please specify the operation that gives you this error. you can also take a look at the setup instructions here - https://drive.google.com/file/d/1l5qg3DIGJ2J9e96xt0fb5lxGk1HxayjQ/view  page 46.

**Comment by gantonjo-tnm** [comment]
Ahhh... Thanks. I missed that document, and as such, I missed the part on installing Arduino IDE. But more importantly, I missed the part where I had to select "Manual" and then select the type of dial and update the libraries. What I did, was install the Dial App and went straight to "Dial-Configuration", created a new "App" configuration and pressed "Create/Save Configuration". The app then crashed, instead of telling me that the libraries were not installed, and prefferably "do you want to install them now?"

**Comment by gantonjo-tnm** [comment]
But now I have another problem. I tried to use the default "Blender" configuration, but when I start Blender and try to control it by the Dial, the Dial App crashes.
```
Application: Ahmsville Dial.exe
Framework Version: v4.0.30319
Description: The process was terminated due to an unhandled exception.
Exception Info: System.IndexOutOfRangeException
   at Ahmsville_Dial.MainWindow.DataReceivedHandler(System.Object, System.IO.Ports.SerialDataReceivedEventArgs)
   at System.IO.Ports.SerialPort.CatchReceivedEvents(System.Object, System.IO.Ports.SerialDataReceivedEventArgs)
   at System.IO.Ports.SerialStream+EventLoopRunner.CallReceiveEvents(System.Object)
   at System.Threading.QueueUserWorkItemCallback.WaitCallback_Context(System.Object)
   at System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   at System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   at System.Threading.QueueUserWorkItemCallback.System.Threading.IThreadPoolWorkItem.ExecuteWorkItem()
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
   at System.Threading._ThreadPoolWaitCallback.PerformWaitCallback()

```

```
Faulting application name: Ahmsville Dial.exe, version: 1.1.0.0, time stamp: 0xd8335704
Faulting module name: KERNELBASE.dll, version: 10.0.19041.1202, time stamp: 0x448a4f5d
Exception code: 0xe0434352
Fault offset: 0x0012b5b2
Faulting process id: 0x3a3c
Faulting application start time: 0x01d7af300459b0aa
Faulting application path: C:\Program Files\Ahmsville Labs\Ahmsville Dial\Ahmsville Dial.exe
Faulting module path: C:\Windows\System32\KERNELBASE.dll
Report Id: 72a44639-5158-4a6c-af51-e0ccbad93bbc
Faulting package full name: 
Faulting package-relative application ID: 

```

**Comment by ahmsville** [comment]
Hi @gantonjo-tnm , blender support is a new feature, so i am still ironing out the bugs, and i have not made an official instruction for the setup yet, but it does work well...so please follow this steps to set it up...

- Install the latest version of the ahmsville dial app (please download again, i just made some changes to the release) - https://github.com/ahmsville/Ahmsville-Dial-App/releases/tag/V1.4
- run the app as administrator and update the library using the update library button.
- plug in your dial and re-upload the configuration.
- locate the installation directory for the ahmsville dial app (C:\Program Files\Ahmsville Labs\Ahmsville Dial)
- find the folder named BLENDER Addon, this folder should contain the blender .py addon that allows blender to work with the dial. 
![snip](https://user-images.githubusercontent.com/44074914/134258160-0fde4cfa-1145-4da2-9cb0-bb6ef46b0423.PNG)

- open blender and install this addon, once installed, click on ctrl + w, or go under view to see the ahmsville dial panel button, 
click on it, and the dial will begin to control your model.
![Untitled](https://user-images.githubusercontent.com/44074914/134258191-809b7fbf-2a71-4e7c-a562-7b32a8357eef.png)

- you can also open the redo panel to make some changes to the settings.

**Comment by gantonjo-tnm** [comment]
Thanks for your answers, @ahmsville.
I tried to install the plugin in my Blender v2.91 but got the following error when I enabled the plugin: 
![image](https://user-images.githubusercontent.com/42378451/134380890-cd572dbf-9cb0-41d8-941e-b3d563a6d82c.png)

**Comment by ahmsville** [comment]
Hi @gantonjo-tnm , this is a missing python module error, the addon requires the pywin32 to work - heres a link to the module - https://pypi.org/project/pywin32/ , to install it specifically for blender please follow the steps in this post - http://www.codeplastic.com/2019/03/12/how-to-install-python-modules-in-blender/ ,i would say use pip (pip install pywin32) in the command prompt to install the module, but as i do not know how python is setup on your system, i cannot guarantee that normal pip install will work, so installing the module specifically for blender is my recommended method.

**Comment by gantonjo-tnm** [comment]
Good evening, @ahmsville.
Well, I am progressing here, managed to install the missing Python libraries in my Blender v2.93. But I am only seeing some kind of action on the macro keys. Dials and tilt are not giving any noticeable actions in my Blender.

When I compile the Arduino sketches, I get the following message:
```
In file included from C:\Users\GunderAntonJohansen\Documents\Arduino\libraries\AhmsvilleDial_v2\examples\Absolute_Variant_main\Absolute_Variant_main.ino:2:0:
C:\Users\GunderAntonJohansen\Documents\Arduino\libraries\FastLED\src/FastLED.h:14:21: note: #pragma message: FastLED version 3.004.000
 #    pragma message "FastLED version 3.004.000"
                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from C:\Users\GunderAntonJohansen\Documents\Arduino\libraries\FastLED\src/controller.h:9:0,
                 from C:\Users\GunderAntonJohansen\Documents\Arduino\libraries\FastLED\src/FastLED.h:47,
                 from C:\Users\GunderAntonJohansen\Documents\Arduino\libraries\AhmsvilleDial_v2\examples\Absolute_Variant_main\Absolute_Variant_main.ino:2:
C:\Users\GunderAntonJohansen\Documents\Arduino\libraries\FastLED\src/pixeltypes.h: In function 'void DeactivateDial(bool)':
C:\Users\GunderAntonJohansen\Documents\Arduino\libraries\FastLED\src/pixeltypes.h:163:11: warning: iteration 12 invokes undefined behavior [-Waggressive-loop-optimizations]
         r = (colorcode >> 16) & 0xFF;
         ~~^~~~~~~~~~~~~~~~~~~~~~~~~~
C:\Users\GunderAntonJohansen\Documents\Arduino\libraries\AhmsvilleDial_v2\examples\Absolute_Variant_main\Absolute_Variant_main.ino:2260:21: note: within this loop
   for (int i = 0; i < NUM_LEDS; i++) { // update macro key leds

```

**Comment by ahmsville** [comment]
Hi, @gantonjo-tnm , are you able to compile and upload configurations to the dial?, remember, you need to plug in the dial via USB to do this.

**Comment by gantonjo-tnm** [comment]
Hi, @ahmsville. Yes, I am able to compile, with the messages I gave you yesterday, and I am able to upload the configuration to the dial via USB. Arduion IDE verifies OK after upload.

```
Sketch uses 111348 bytes (42%) of program storage space. Maximum is 262144 bytes.
Global variables use 9032 bytes (27%) of dynamic memory, leaving 23736 bytes for local variables. Maximum is 32768 bytes.
Atmel SMART device 0x10010005 found
Device       : ATSAMD21G18A
Chip ID      : 10010005
Version      : v2.0 [Arduino:XYZ] Nov 18 2020 00:44:43
Address      : 8192
Pages        : 3968
Page Size    : 64 bytes
Total Size   : 248KB
Planes       : 1
Lock Regions : 16
Locked       : none
Security     : false
Boot Flash   : true
BOD          : true
BOR          : true
Arduino      : FAST_CHIP_ERASE
Arduino      : FAST_MULTI_PAGE_WRITE
Arduino      : CAN_CHECKSUM_MEMORY_BUFFER
Erase flash
done in 1.008 seconds

Write 111348 bytes to flash (1740 pages)
[==============================] 100% (1740/1740 pages)
done in 0.839 seconds

Verify 111348 bytes of flash with checksum.
Verify successful
done in 0.075 seconds
CPU reset.

```

I have for instance tried to add "Mouse" to the Space Nav for the Default profile, but only action that is happening when usin the dial is volume up/down and start/stop media. The mouse is not moving when tilting or sliding the space nav.

**Comment by ahmsville** [comment]
hi @gantonjo-tnm are you using the dial in wired or wireless mode?

**Comment by gantonjo-tnm** [comment]
Hi, @ahmsville. I have trided wireless, but then the Dial App crashes. So, all tests I have done so far are through USB cable.

**Comment by ahmsville** [comment]
Hi, @gantonjo-tnm , you can test if the spacenav is working as it should by uploading the sample code located under examples, SpaceNavigator.

![sp1](https://user-images.githubusercontent.com/44074914/134711332-ff6ab0bb-f82b-481f-9f96-bf61909d21a3.png)

upload this sample sketch to your dial, and open the serial monitor, then move the spacenav around to see if you get the movement data.
you should see something like this...
![sp2](https://user-images.githubusercontent.com/44074914/134711484-ab729c77-9ea9-4fc1-aac4-d06f73143174.png)

Also remember that the app is context sensitive, so if you try to use the spacenav for blender when blender is not in the foreground, it will not work, also make sure to use ctrl + w whenever you open blender to launch the addon.



for the wireless adapter, i think you just need to update the code, you can do this by plugging in the wireless adapter and uploading the sketch found in examples, AhmsvilleDial v2, wireless adapter v1.1.
![sp3](https://user-images.githubusercontent.com/44074914/134712021-1c51bfa1-6e43-40f3-898c-d124000d0eee.png)

**Comment by gantonjo-tnm** [comment]
Hi again, @ahmsville. 
I have tried to update the wireless adapter as indicated and the dial successfully connect to the wireless adapter.
I have also uploaded the spacenav lib and could see the actions as you mention.

So, I have tried to create a new config for the Absolute version, adding some features to both Default and Blender configurations. The only one that works (both via USB and wireless) is volume/up in Default, and MK2 in Blender. I do not see any lights related to MK-keys (I guess there should be LEDs under each MK-key?)

Maybe the lack of lights is due to these messages when compiling "Absolute_variant_main"?
```
In file included from C:\Users\GunderAntonJohansen\Documents\Arduino\libraries\AhmsvilleDial_v2\examples\Absolute_Variant_main\Absolute_Variant_main.ino:2:0:
C:\Users\GunderAntonJohansen\Documents\Arduino\libraries\FastLED\src/FastLED.h:14:21: note: #pragma message: FastLED version 3.004.000
 #    **pragma message "FastLED version 3.004.000"
                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~**
In file included from C:\Users\GunderAntonJohansen\Documents\Arduino\libraries\FastLED\src/controller.h:9:0,
                 from C:\Users\GunderAntonJohansen\Documents\Arduino\libraries\FastLED\src/FastLED.h:47,
                 from C:\Users\GunderAntonJohansen\Documents\Arduino\libraries\AhmsvilleDial_v2\examples\Absolute_Variant_main\Absolute_Variant_main.ino:2:
C:\Users\GunderAntonJohansen\Documents\Arduino\libraries\FastLED\src/pixeltypes.h: In function 'void DeactivateDial(bool)':
C:\Users\GunderAntonJohansen\Documents\Arduino\libraries\FastLED\src/pixeltypes.h:163:11: warning: iteration 12 invokes undefined behavior [-Waggressive-loop-optimizations]
         r = (colorcode >> 16) & 0xFF;
         ~~^~~~~~~~~~~~~~~~~~~~~~~~~~
C:\Users\GunderAntonJohansen\Documents\Arduino\libraries\AhmsvilleDial_v2\examples\Absolute_Variant_main\Absolute_Variant_main.ino:2260:21: note: within this loop
   **for (int i = 0; i < NUM_LEDS; i++) { // update macro key leds
Sketch uses 111340 bytes (42%) of program storage space. Maximum** is 262144 bytes.
Global variables use 9032 bytes (27%) of dynamic memory, leaving 23736 bytes for local variables. Maximum is 32768 bytes.

```

Also, the Dial App crashes if I use Wireless and forget to activate the Dial plugin in Blender.

**Comment by gantonjo-tnm** [comment]
Ahhh, by the way. I tested the MK-key sketch and see the correct reaction in the serial monitor when pressing the 5 MK keys.

**Comment by ahmsville** [comment]
@gantonjo-tnm ,i think its a library problem, please check your arduino library folder to make sure you don't have multiple versions of the FastLED library. the version bundled with the app is 3.003.003.
![sp4](https://user-images.githubusercontent.com/44074914/134739770-174775d8-81a1-496b-838c-49d3683e68b8.PNG)

with regards the app crashing when using wireless, its probably a bug, i'll look into fixing that right away.

**Comment by gantonjo-tnm** [comment]
I have deleted all folders in my Arduino folder and let the Dial App reinstall all from scratch:
![image](https://user-images.githubusercontent.com/42378451/134744297-b81ffc10-01ab-47ba-a056-8b4b3dcd5c9f.png)
Then I uploaded the Abolute Version to the dial and still no luck in getting things to work as intended. I am sure that e.g. FastLED is version 3.3.3 as bundled with your code. I have not updated to latest versions. Same error when compiling.

What I now notice, after looking at the installation description, is that only the 5 first LEDs clock wise from the USB port, and the one below the touch sensor, are lit. Well, the green led next to the power button is also lit. I guess there are at least 2 more LEDs in the base to be litt and the 5 MK button LEDs, right?
